mock/mock.spec

```
@@ -1,6 +1,19 @@
 %bcond_with lint
 %bcond_without tests
 
+# In modern distributions with RPM v4.19+ (e.g., Fedora 39+), the %%pre
+# scriptlet for creating users/groups is no longer necessary due to the built-in
+# sysusers feature.  To enable it, we simply include the `mock.conf` file in the
+# package payload so that RPM knows what actions to take.  For older
+# distributions, however, %%pre is needed.  On some of those older distros the
+# %%sysusers_*_compat macros could help (but not everywhere), so turns out it is
+# just way easier and to expand those macros and hardcode them in this specfile.
+%if 0%{?rhel} && 0%{?rhel} <= 9
+%bcond_without sysusers_compat
+%else
+%bcond_with sysusers_compat
+%endif
```

This needs to be flipped. We only know for _sure_ that newer Fedora and RHEL support it, and everything else we don't know.

---

```suggestion
%if 0%{?fedora} || 0%{?rhel} >= 10
%bcond_with sysusers_compat
%else
%bcond_without sysusers_compat
%endif
```

---

Or alternatively if we assume everyone is going to get it, then we need something like this:

```suggestion
%if (0%{?rhel} && 0%{?rhel} < 10) || (0%{?mageia} && 0%{?mageia} < 10) || (0%{?suse_version} && 0%{?suse_version} < 1660)
%bcond_without sysusers_compat
%else
%bcond_with sysusers_compat
```

---

the first suggestion accepted, thank you